src/Models/Context/BehaviorFormContext/BehaviorFormContext.tsx

```diff
@@ -200,6 +209,47 @@ export const BehaviorFormContextReducer: (
                 );
                 break;
             }
+            case BehaviorFormContextActionType.FORM_BEHAVIOR_VISUAL_RULE_ADD_OR_UPDATE: {
+                let draftVisualRules = draft.behaviorToEdit.visuals.filter(
+                    ViewerConfigUtility.isVisualRule
+                );
+                if (action.payload.visualRule.id) {
+                    draftVisualRules = AddOrUpdateListItemByFilter(
+                        draftVisualRules,
+                        action.payload.visualRule,
+                        (x) => x.id === action.payload.visualRule.id,
+                        logDebugConsole
+                    );
+                } else {
+                    // It is not possible to receive a new visual rule without an id, this just considers update scenario for legacy data
```

Comment:
In create flow we always have an id, since it will be required by the UI in the form. Legacy data exists that doesn't contain an id, it will only be updated never created.